### ISSUE TITLE
Cannot display a previously saved map (raster)

### ISSUE THREAD

**Issue by Nimn** [Rolisteam/rolisteam#76]
<p align='center'><img src="https://raw.githubusercontent.com/Rolisteam/rolisteam/master/resources/logo/1000-rolisteam.png"/></p>

#### Configuration
*  Linux (Debian Stretch) v1.8.2 (compiled from sources)

#### Rolisteam Setting
* GM and Server

#### What did you do?
- Start rolisteam
- Open a previously saved map

#### What did you expect to see?
The map! 

#### What did you get instead?
No new window is opened. But strangely if a player is connected the player can see the map. As a workaround I can open a vectorial map without any problem.

**Comment by obiwankennedy** [comment]
Did you load created by rolisteam 1.8.2 ? or from previous version ? Can I have a copy of that map ?

**Comment by Nimn** [comment]
The map was created with rolisteam 1.8.2. In fact I can not load any map I create.
Here a very simple map (only a circle is drawn): 
[test-map.zip](https://github.com/Rolisteam/rolisteam/files/1691725/test-map.zip)

**Comment by obiwankennedy** [comment]
It works fine for me. I can load it from the menu, as drag and drop and even if I checked the «hidden» on the map loader wizard.

**Comment by Nimn** [comment]
5.7.1: https://packages.debian.org/stretch/qtbase5-dev

**Comment by Nimn** [comment]
For information: I was able to reproduce the same behavior on another computer with a brand new Debian Stretch installation.

**Comment by obiwankennedy** [comment]
I compiled from source on ubuntu and I can't reproduce the issue. 
Did you try to load any jpg image into rolisteam ?
I'm thinking perhaps you don't have the right image backend. So your rolisteam is not able to load/display jpg data. 

Maps are saved as 3 jpg images.
Does the window have the right title or has the right number of NPC on it ?

**Comment by Nimn** [comment]
I can load jpg in rolisteam.

I don't use many qt applications and compared to Ubuntu, Debian is shipped with less libraries by default. So I also think it could be a missing dependency. For instance to make the music works I have to install qtgstreamer-plugins-qt5 package.

Otherwise I have made another test by loading a scenario with a map. The map appears in the resources list and do not load on double click but I have an explicit message: "Format de fichier non supporté". Furthermore I have a message on the console where I launched rolisteam: "QFSFileEngine::open: No file name specified".

I am trying to install (randomly) packages to see if it can solve this issue.

**Comment by Nimn** [comment]
No chance with installation of new packages.

I have played a little bit with the source code (v1.8.2). It's been years since I have not programmed in C++ so my understanding of the code is a little bit laborious. With this patch I am now able to load map from a scenario (not directly from the open menu):

```
--- ./map/mapframe.cpp	2018-02-04 21:46:39.452632112 +0100
+++ map/mapframe.cpp.origin	2018-02-04 20:10:13.386085040 +0100
@@ -177,7 +177,8 @@
 
 bool MapFrame::readFileFromUri()
 {
-    if((NULL!=m_uri)&&(!m_uri->getData().isEmpty()))
+
+    if((NULL!=m_uri)&&(!m_uri->getData().isEmpty())&&(m_uri->getUri().isEmpty()))
     {//load from uri data
         QByteArray array = m_uri->getData();
         QDataStream in(&array,QIODevice::ReadOnly);
```

I think this is only a side effect and not a real patch but maybe it could help on debug?

**Comment by obiwankennedy** [comment]
To test about your problem, I compile myself code source of the 1.8.2 from source archive and it worked like a charm. 

My next step will be to test on virtual machin and try to debug it.

At least I can tell you mapframe.cpp is the class dedicated to manage the ClerverURI of the map. 
If you want to know where the problem occur. You may add some code like this:

```
//at the top of the file
#include <QDebug>


//in code
qDebug() << "here 1";
qDebug() << "here 2";
...
```


Given your problem, It is not able to do something. As the application does not crash. Rolisteam may be aware of the failure. 

it may look like that:

```
if(object->doSomething())
{
// success
}
else
{
//fail
}
```

If you're able to find out the exact line which fail. It may help.
______________________________

Another way to know if something is wrong. You can perform a diagnostics. 
On Ubuntu, this causes the application to crashes. Perhaps on Debian it will works.
If it does, copy/paste the result here.

To do so: 
* Open Rolisteam 
* Open Preferences
* Go to the last tab (diag)
* Start diag

The diagnostics should display all supported fonts, images formats and audio formats.

**Comment by obiwankennedy** [comment]
I start to create a virtual machine to compile rolisteam on Debian. It is not finished yet but I hope to test it tonight.

**Comment by obiwankennedy** [comment]
Something is wrong with my virtual machin. I can't make debian install on it. I may try another way.

**Comment by obiwankennedy** [comment]
Ok! I think I have the answer.

Could you add this line:

`return true;`

At the line 176 of the file: mapframe.cpp ?

**Comment by Nimn** [comment]
Yes! It fixes part of the problem.
With this patch I am able to open a previously saved map from the menu.

But I am still not able to open a map from a previously saved scenario. Same message "Format de fichier non supporté" and in the console used to open rolisteam I have the message: "QFSFileEngine::open: No file name specified"

**Comment by obiwankennedy** [comment]
I can't reproduce this other problem. Could you send me a scenario with this issue ?

**Comment by Nimn** [comment]
A scenario with this issue:
[scenario-test.zip](https://github.com/Rolisteam/rolisteam/files/1775045/scenario-test.zip)

By the way I have another issue which may be related (or not). On the recent items menu, the names are not displayed:
![2018-03-rolisteam](https://user-images.githubusercontent.com/412978/36898973-64b87cfa-1e1d-11e8-9e67-337d1943b46f.jpg)

**Comment by obiwankennedy** [comment]
The name issue on recent files is known and already fixed for the v1.9.

**Comment by obiwankennedy** [comment]
Ok, I think I got it. 
The scenario stores only the path of to the media. So In your scenario the map is expected to be found at: /home/etienne/Documents/jdr/rolisteam/chtulhu/test/test.pla 

Does a plan exist there ? Because I got the same error message than you and on my computer I have no file at that place.
